azurerm/resource_arm_storage_account.go

```diff
@@ -829,6 +829,9 @@ func expandStorageAccountBypass(networkRule map[string]interface{}) storage.Bypa
 }
 
 func flattenStorageAccountNetworkRules(input *storage.NetworkRuleSet) []interface{} {
+	if len(*input.IPRules) == 0 && len(*input.VirtualNetworkRules) == 0 {
```

Comment:
which means this could become

```
if input == nil {
  return []interface{}{}
}
```

Reply:
and we could make this: https://github.com/terraform-providers/terraform-provider-azurerm/pull/1613/files#diff-f4d2502d01b7fb77deb7bb221025fb51R845

```
ipRules := make([]interface{}, 0)

if input != nil {
  for i, ipRule := range *input {
    if range := ipRule.IPAddressOrRange; range != nil {
      ipRules[i] = *range
    }
  }
}

return ipRules
```

(and same for the other two sub-functions)